client/src/main/java/hudson/plugins/swarm/SwarmClient.java

```diff
@@ -653,13 +671,48 @@ private static String hash(File remoteFsRoot) {
     }
 
     public void exitWithStatus(int status) {
+        if (prometheusServer != null) {
+            prometheusServer.stop(1);
+        }
         System.exit(status);
     }
 
     public void sleepSeconds(int waitTime) throws InterruptedException {
         Thread.sleep(waitTime * 1000);
     }
 
+    private void startPrometheusService(int port) {
+        logger.fine("Starting Prometheus service on port " + port);
+        PrometheusMeterRegistry prometheusRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+        // Add some standard metrics to the registry
+        new ClassLoaderMetrics().bindTo(prometheusRegistry);
+        new FileDescriptorMetrics().bindTo(prometheusRegistry);
+        new JvmGcMetrics().bindTo(prometheusRegistry);
+        new JvmHeapPressureMetrics().bindTo(prometheusRegistry);
+        new JvmMemoryMetrics().bindTo(prometheusRegistry);
+        new JvmThreadMetrics().bindTo(prometheusRegistry);
+        new ProcessorMetrics().bindTo(prometheusRegistry);
+        new UptimeMetrics().bindTo(prometheusRegistry);
+
+        try {
+            prometheusServer = HttpServer.create(new InetSocketAddress(port), 0);
+            prometheusServer.createContext("/prometheus", httpExchange -> {
+                String response = prometheusRegistry.scrape();
+                byte[] responseContent = response.getBytes(StandardCharsets.UTF_8);
+                httpExchange.sendResponseHeaders(200, responseContent.length);
+                try (OutputStream os = httpExchange.getResponseBody()) {
+                    os.write(responseContent);
+                }
+            });
+
+            new Thread(prometheusServer::start).start();
+        } catch (IOException e) {
+            logger.severe("Failed to start Prometheus service: " + e.getMessage());
+            throw new RuntimeException(e);
```

Comment:
In Java 8 and later, `UncheckedIOException` can be used as a more specific exception type when you want to rethrow an `IOException`.

```suggestion
            throw new UncheckedIOException(e);
```